src/go/rpk/pkg/cli/cmd/cluster/txn/describe.go

```diff
@@ -0,0 +1,147 @@
+
+By default, all topics in a transaction are merged into one line. To print a
+row per topic, use --format=long. To include partitions with topics, use
+--print-partitions,
+
+If no transactional IDs are requested, all transactional IDs are printed.
+`,
+
+		Run: func(cmd *cobra.Command, txnIDs []string) {
+			p := config.ParamsFromCommand(cmd)
+			cfg, err := p.Load(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			adm, err := kafka.NewAdmin(fs, p, cfg)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer adm.Close()
+
+			described, err := adm.DescribeTransactions(cmd.Context(), txnIDs...)
+			out.HandleShardError("DescribeTransactions", err)
+
+			headers := []string{
+				"coordinator",
+				"transactional-id",
+				"producer-id",
+				"producer-epoch",
+				"state",
+				"start-timestamp",
+				"timeout",
+			}
+			common := func(x kadm.DescribedTransaction) []interface{} {
+				return out.StructFields(struct {
+					Coordinator    int32
+					TxnID          string
+					ProducerID     int64
+					ProducerEpoch  int16
+					State          string
+					StartTimestamp string
+					Timeout        time.Duration
+				}{
+					x.Coordinator,
+					x.TxnID,
+					x.ProducerID,
+					x.ProducerEpoch,
+					x.State,
+					time.UnixMilli(x.StartTimestamp).Format(rfc3339Milli),
+					time.Duration(x.TimeoutMillis) * time.Millisecond,
+				})
+			}
+
+			switch format {
+			case "short", "text":
+				tw := out.NewTable(append(headers, "topics")...)
+				defer tw.Flush()
+				for _, x := range described.Sorted() {
+					// Without partitions, we format like "foo,bar,baz".
+					// With partitions, we format "foo[0;1;2],bar[1;2;3]
+					var topics string
+					if printPartitions {
+						var ts []string
+						for _, t := range x.Topics.Sorted() {
+							var ps []string
+							for _, p := range t.Partitions {
+								ps = append(ps, strconv.Itoa(int(p)))
+							}
+							ts = append(ts, fmt.Sprintf("%s[%s]", t.Topic, strings.Join(ps, ";")))
+						}
+						topics = strings.Join(ts, ",")
+					} else {
+						ts := x.Topics.Topics()
+						sort.Strings(ts)
+						topics = strings.Join(ts, ",")
+					}
+					tw.Print(append(common(x), topics))
+				}
+
+			case "long":
+				if printPartitions {
+					tw := out.NewTable(append(headers, "topic", "partition")...)
+					defer tw.Flush()
+					for _, x := range described.Sorted() {
+						for _, t := range x.Topics.Sorted() {
+							for _, p := range t.Partitions {
+								tw.Print(append(common(x), t.Topic, p))
+							}
+						}
+					}
+				} else {
+					tw := out.NewTable(append(headers, "topic")...)
+					defer tw.Flush()
+					for _, x := range described.Sorted() {
+						for _, t := range x.Topics.Sorted() {
+							tw.Print(append(common(x), t.Topic))
+						}
+					}
+				}
+			default:
+				out.Die("unrecognized format %q", format)
+			}
+		},
+	}
+
+	cmd.Flags().StringVar(&format, "format", "text", "Output format (short, long)")
```

Comment:
I think we should leave `format` flag only for json, yaml, toml, etc... like in the other commands.

for this particular command wdyt of only have a flag `--long` or something like that instead of a format?

Reply:
I'm not sure about that, having a second flag that changes the format could be a bit weird. I was thinking short is the same as text (hence the switch above), and we could use short/long in cases where most people only care about short output _but_ we have the option for some super detailed format. `long` could be seen as detailed output similar to json, but in a text based format.

(also I was thinking this could mirror kubectl's `-o wide`, vs the normal short output)

Reply:
But using `--format` will leave us with the mismatch between the format flag in other commands vs this one.

What about detailed? We have `--detailed` in `rpk topic list`, we also have --print-summary in `group describe`.

Reply:
Summarizing DM's: fine to leave for now, also we may want to port `--detailed` in `rpk topic list` to using `--format long`.

I'll also add `wide` as an alias for `long` for people more familiar with kubectl.